homeassistant/helpers/entity.py

```diff
@@ -869,8 +873,11 @@ def _async_write_ha_state(self) -> None:
             self._context_set = None
 
         try:
+            # The get guards against integrations not using an EntityComponent and can
+            # be removed in HA Core 2024.1
+            entity_info = entity_sources(hass).get(self.entity_id)
```

Comment:
Maybe set this as `self._entity_sources` when added to hass to avoid the function call and lookup here

Reply:
Well I guess its created here so we could just hold a reference to it instead in https://github.com/home-assistant/core/pull/100239/files#diff-9243613f337d1edb46e223da7399bbbca469d21082dc2114c0cf65a5d6a05959R1086